### ISSUE TITLE
Allow default resource limits to be specified in the artifact definition

### ISSUE THREAD

**Issue by scudette** [Velocidex/velociraptor#1011]
Currently resource limits are hard coded but some artifact need longer time or more uploaded bytes. By having these in the artifact definition we can tailor those limits to the right artifacts.

**Comment by predictiple** [comment]
The `timeout` value doesn't seem to be applied

![dex_offline yaml - dex-artifacts - Visual Studio Code_802](https://user-images.githubusercontent.com/9843422/114387159-69f31180-9b92-11eb-846b-6e3be827f1cd.png)

result
![Velociraptor Response and Monitoring - Brave_801](https://user-images.githubusercontent.com/9843422/114387181-70818900-9b92-11eb-8388-d8a251ba9230.png)

Also it would be nice if the "New Collection: Specify Resources" screen showed the values if specified, but I guess that might be tricky if multiple artifacts are selected?

**Comment by scudette** [comment]
Timeout is actually applied for each query (i.e. artifact + source) in a collection.

The GUI probably needs to show something else here (probably "default" rather than 600).

When specifying the timeout in the resources tab then it overrides the timeout in the artifact definition. But if you don't specify then it will be taken as the artifact default, or if there is no default then 600.

You should be able to see the correct timeout set in the request tab.

**Comment by predictiple** [comment]
Ah thanks, yes it does show the correct timeout value in the Requests tab.

**Comment by predictiple** [comment]
The resources settings don't seem to apply at all when used in a server artifact.

the artifact definition specifies:
```yaml
resources:
  timeout: 666666
  max_rows: 999999999
  max_upload_bytes: 99999999999
```

Incidentally on the New Collection tab the timeout value is shown but not the others:

![Screenshot from 2021-04-15 14-22-14](https://user-images.githubusercontent.com/9843422/114868873-ae7ee700-9df6-11eb-95d2-d16ce5323e3d.png)



Running the collection, the Requests tab shows:
```yaml
"max_row": "1000",
"timeout": "666666"
```

and the collection times out after 600 seconds:

![Screenshot from 2021-04-15 14-22-14](https://user-images.githubusercontent.com/9843422/114869488-52689280-9df7-11eb-83e0-35103bad5ffe.png)

It would be handy if the resources settings were applied because my use case is taking a (typically large) collection and forwarding it to Elastic on-demand. The server artifact is a simple one and very similar to `Elastic.Flows.Upload`

**Comment by predictiple** [comment]
forgot to mention, the above testing is with
version: 0.5.8-rc1
commit: cf9311b2

**Comment by predictiple** [comment]
It seems like the timeout value is ignored for server artifacts, even if entered manually on the Specify Resources tab. Even though the Requests tab shows the correct timeout value, the collection always times out at 600 seconds.

**Comment by predictiple** [comment]
Testing with a trivial long-running server artifact confirms that the 600sec timeout is hardcoded. Neither the resources setting in the artifact, nor the Specify Resources tab overrides the timeout (which makes one wonder why that page even displays a Max Execution Time setting).

That means post-processing tasks longer than 10 minutes are impossible.

**Comment by scudette** [comment]
Yes that is correct, as can be seen here 
https://github.com/Velocidex/velociraptor/blob/ce78aecbde9699780b2d4e7ce08a8798e1f82f15/services/server_artifacts/server_artifacts.go#L457

The timeout is fixed.

**Comment by predictiple** [comment]
Is this likely to change?

It doesn't seem improbable that post-processing tasks could take longer than client collection tasks, especially given that the client collection timeout can be arbitrarily long.

**Comment by scudette** [comment]
Yes - this is clearly an oversight :-). Let's open another issue for it.

**Comment by scudette** [comment]
Should be fixed by #1020

**Comment by predictiple** [comment]
Thanks for attending to this so quickly. I really appreciate it. I've tested successfully with the new build.

One thing that's interesting is that the Requests tab shows a max_row of 1000 even though my artifact specifies max_row**s** (note the 's') of 999999999.

![Screenshot from 2021-04-16 19-13-25](https://user-images.githubusercontent.com/9843422/115061010-da7a9500-9ee8-11eb-8349-6e03bfd1fe20.png)

The collection (server artifact) didn't stop at 1000 rows though, so it does seem to be using the value defined in the artifact. Possibly there are some discrepancies with max_row vs. max_row**s**, or maybe max_row is another setting that has some other purpose that I'm just not aware of.

**Comment by scudette** [comment]
Server artifacts don't enforce a row limit at the moment. Do you think they
should?

The main reason for the client side row limits is the large uploads but
this does not really apply to the server artifacts.

**Comment by predictiple** [comment]
I don't think there should be any limits at all, but it's working perfectly now with huge limits :nerd_face: 

With the max_row / max_rows it just seems potentially misleading if it displays a limit when there is none, and it also seems like max_row vs. max_row**s** is possibly an older and a newer name for the same setting that could cause trouble in future if one doesn't notice the missing 's'.

**Comment by scudette** [comment]
Should be fixed in 0.5.9-rc1
